provider/mqtt/client.go

```diff
@@ -49,7 +53,7 @@ type Option func(*paho.ClientOptions)
 const secure = "tls://"
 
 // NewClient creates new Mqtt publisher
-func NewClient(log *util.Logger, broker, user, password, clientID string, qos byte, insecure bool, opts ...Option) (*Client, error) {
+func NewClient(log *util.Logger, broker, user, password, clientID string, qos byte, insecure bool, caCertString string, clientCertString string, clientKeyString string, opts ...Option) (*Client, error) {
```

Comment:
Feels a bit on-goish to append String to the var names?

Reply:
Might also be nice- to avoid parameter proliferation, to implement Ca and client certs as options rather than additional parameters.

Reply:
I've renamed the strings for now but I'll have a look at the options now

Reply:
I have tried building the TLS config in `setup.go` and then passing it in the options, however `helper.go` and `registry.go` also call the `RegisteredClient` function, without any way to pass options. Is this intended? 
If I had to implement the options there too, I'd need to build the TLS config there as well and we'd end up with code duplication. Or am I not seeing something?